ExampleMod/Content/Items/Weapons/ExampleWhip.cs

```diff
@@ -0,0 +1,44 @@
+﻿using ExampleMod.Content.Projectiles;
+using Terraria;
+using Terraria.GameContent.Creative;
+using Terraria.ID;
+using Terraria.ModLoader;
+
+namespace ExampleMod.Content.Items.Weapons
+{
+	public class ExampleWhip : ModItem
+	{
+		public override void SetStaticDefaults() {
+			ItemID.Sets.SummonerWeaponThatScalesWithAttackSpeed[Type] = true; // This makes the item be affected by the user's melee speed.
+			CreativeItemSacrificesCatalog.Instance.SacrificeCountNeededByItemId[Type] = 1;
+		}
+
+		public override void SetDefaults() {
+			// Item.DefaultToWhip(ModContent.ProjectileType<ExampleWhipProjectile>(), 20, 2, 4); // Call this method to quickly set the whip's properties.
```

Comment:
I can't tell which of these values 'I' would or wouldn't need to set if I were to use this.

Maybe consider having two separate SetDefaults() methods, one of them commented out, that shows how to use this?
Further, it would be helpful to indicate which of the following values should be adjusted for 'my' own implementation, be it via grouping or comments.